cmake/sdlchecks.cmake

```diff
@@ -817,8 +817,7 @@ endmacro()
 macro(CheckEGL)
   if (SDL_OPENGL OR SDL_OPENGLES)
     pkg_check_modules(EGL egl)
-    string(REPLACE "-D_THREAD_SAFE;" "-D_THREAD_SAFE=1;" EGL_CFLAGS "${EGL_CFLAGS}")
```

Comment:
It looks like you removed this modification of EGL_CFLAGS. Was that intentional?

Reply:
Yes.
Changing a '-Dmacro' to 'Dmacro=1' does make the cmake error also go away.
My interpretation of the line is that this was put here to make that error go away when building against an egl implementation which did set -D_THREAD_SAFE in its egl.pc file.
This is fixed not only for _THREAD_SAFE but for any other macro egl.pc would asked to be part of the CFLAGS by putting EGL_CFLAGS in the CMAKE_REQUIRED_DEFINITIONS variable. It will then even be passed to the C compiler and do what it is intended for rather than being silently ignored.

Reply:
Okay, thanks for clarifying.